### ISSUE TITLE
Support MMDetection Models

### ISSUE THREAD

**Issue by ffcai** [eto-ai/rikai#520]
We want to run models in [MMDetection](https://github.com/open-mmlab/mmdetection) on Spark with Rikai.

Here is [an example of image inference](https://mmdetection.readthedocs.io/en/v2.20.0/get_started.html#verification)

Had some discussion with @eddyxu , so we can:

1. develop a `mmdetection flavor`
2. wrap all the `mmdetection` code into a `torch.nn.Module` class

1 is not maintainable if we keep making each model a flavor, 2 is more general but need a try 

any thoughts?

**Comment by darcy-shen** [comment]
How about maintaining rikai-mmdetection like [rikai-yolov5](https://github.com/Tubitv/rikai-yolov5)?

**Comment by ffcai** [comment]
> How about maintaining rikai-mmdetection like [rikai-yolov5](https://github.com/Tubitv/rikai-yolov5)?

so do we need to wrap the [inference_detector](https://github.com/open-mmlab/mmdetection/blob/master/mmdet/apis/inference.py#L89-L152) code into rikai-mmdetection? How much time to implement it?

**Comment by darcy-shen** [comment]
To write a `rikai-xyz`, one need first to know rikai and xyz both. And then, just re-rewrite the code snippet in xyz in three parts in rikai:
+ pre_processor
+ codegen.generate_udf
+ post_processor

**Comment by darcy-shen** [comment]
It takes about several hours to split one python function to three python functions, and you do not need to create code that actually does not exist.

**Comment by eddyxu** [comment]
i don't really suggest writing a new flavor (i.e., a new `codegen.generate_udf`).  Lets try to make it work with torch runner first?  If the interface of the current torch runner does not fit, we should make the interface more flexible.

The reason is that maintaining a few core runners (torch , tf, scikit-learn) can make the team focus on optimizing the performance for each and introduce some changes between Spark scala and these runners without worrying about breaking userland code. We do have a great roadmap to change/optimize how the models are executed.

If we went to the `per-non-standard-model-per-flavor` route, the project will become unmanageable, and rikai must be stuck with the current design and can not rapidly evolve. 

TLDR, let's work together to try to make the existing runner / flavor more flexible?

**Comment by darcy-shen** [comment]
#525 seems to be a good solution for me.

**Comment by ffcai** [comment]
[mmdet model detectors](https://mmdetection.readthedocs.io/en/latest/api.html#module-mmdet.models.detectors) worked perfectly in Rikai. I have tested with https://github.com/ffcai/rikai/commit/62f374a9c765bd5ab41934f46fa3111dbb547e61
[Faster R-CNN](https://mmdetection.readthedocs.io/en/latest/model_zoo.html#faster-r-cnn) ✅
[Cascade R-CNN](https://github.com/open-mmlab/mmdetection/blob/master/configs/cascade_rcnn) ✅ 
...

I'll send a PR after some refine work
